### PR TITLE
UCS/UCT/UCP: Fixing compilation warning

### PR DESCRIPTION
**Pull request by shamisp**
## What
Fixing build issue

## Why ?
GCC 9.3.1 on Fedora 30 (aarch64) fails with multiple warnings
complaining about usage of uninitialized variables.
Configure command line: `configure --enable-compiler-opt=g`

### CODE REVIEW

**Comment by shamisp** [comment]
ok to test

File: src/ucp/proto/rndv.c [hunk 1]
```diff
@@ -957,7 +957,7 @@ static void ucp_rndv_do_rkey_ptr(ucp_request_t *rndv_req, ucp_request_t *rreq,
     ucp_ep_h ep                      = rndv_req->send.ep;
     const ucp_ep_config_t *ep_config = ucp_ep_config(ep);
     ucp_worker_h worker              = rreq->recv.worker;
-    ucp_md_index_t dst_md_index;
+    ucp_md_index_t dst_md_index = 0;
```
Comment (yosefe) [inline src/ucp/proto/rndv.c:960]: align = on column

File: src/ucp/tag/eager_rcv.c [hunk 2]
```diff
@@ -156,7 +156,7 @@ ucp_eager_common_middle_handler(ucp_worker_t *worker, void *data, size_t length,
 {
     ucp_eager_middle_hdr_t *hdr = data;
     ucp_tag_frag_match_t *matchq;
-    ucp_recv_desc_t *rdesc;
+    ucp_recv_desc_t *rdesc = NULL;
```
Comment (yosefe) [inline src/ucp/tag/eager_rcv.c:159]: align = on column and move to line 158 (initialized vars should come first)

File: src/ucs/async/async.c [hunk 3]
```diff
@@ -178,7 +178,7 @@ static ucs_status_t ucs_async_handler_add(int min_id, int max_id,
     ucs_async_handler_t *handler_from_hash;
     int hash_extra_status;
     ucs_status_t status;
-    khiter_t hash_it;
+    khiter_t hash_it = 0;
```
Comment (yosefe) [inline src/ucs/async/async.c:181]: move to line 178 (initialized vars should come first)

File: src/ucs/sys/sock.c [hunk 4]
```diff
@@ -652,7 +652,7 @@ int ucs_sockaddr_cmp(const struct sockaddr *sa1,
 {
     int result          = 1;
     ucs_status_t status = UCS_OK;
-    uint16_t port1, port2;
+    uint16_t port1 = 0, port2 = 0;
```
Comment (yosefe) [inline src/ucs/sys/sock.c:655]: align = on column

File: src/uct/ib/dc/dc_mlx5_ep.c [hunk 5]
```diff
@@ -272,7 +272,7 @@ uct_dc_mlx5_ep_short_dm(uct_dc_mlx5_ep_t *ep, uct_rc_mlx5_dm_copy_data_t *cache,
                         uint64_t rdma_raddr, uct_rkey_t rdma_rkey)
 {
     uct_dc_mlx5_iface_t *iface = ucs_derived_of(ep->super.super.iface, uct_dc_mlx5_iface_t);
-    uct_rc_iface_send_desc_t *desc;
+    uct_rc_iface_send_desc_t *desc = NULL;
```
Comment (yosefe) [inline src/uct/ib/dc/dc_mlx5_ep.c:275]: align = on column

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 6]
```diff
@@ -583,7 +583,7 @@ static UCS_F_ALWAYS_INLINE void
 uct_rc_mlx5_handle_rndv_fin(uct_rc_mlx5_iface_common_t *iface, uint32_t app_ctx)
 {
     int found;
-    void *rndv_comp;
+    void *rndv_comp = NULL;
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h:586]: move to line 585 (initialized vars should come first)

File: src/uct/ib/rc/accel/rc_mlx5_ep.c [hunk 7]
```diff
@@ -132,7 +132,7 @@ uct_rc_mlx5_ep_short_dm(uct_rc_mlx5_ep_t *ep, uct_rc_mlx5_dm_copy_data_t *cache,
 {
     uct_rc_mlx5_iface_common_t *iface = ucs_derived_of(ep->super.super.super.iface,
                                                        uct_rc_mlx5_iface_common_t);
-    uct_rc_iface_send_desc_t *desc;
+    uct_rc_iface_send_desc_t *desc = NULL;
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_ep.c:135]: align = on column

File: src/uct/tcp/tcp_ep.c [hunk 8]
```diff
@@ -1481,7 +1481,7 @@ ucs_status_t uct_tcp_ep_am_zcopy(uct_ep_h uct_ep, uint8_t am_id, const void *hea
     uct_tcp_ep_t *ep           = ucs_derived_of(uct_ep, uct_tcp_ep_t);
     uct_tcp_iface_t *iface     = ucs_derived_of(uct_ep->iface, uct_tcp_iface_t);
     uct_tcp_ep_zcopy_tx_t *ctx = NULL;
-    size_t payload_length;
+    size_t payload_length = 0;
```
Comment (yosefe) [inline src/uct/tcp/tcp_ep.c:1484]: align = on column

**Comment by shamisp** [comment]
@yosefe good to go ?

**Pull request merged**